jujugui/static/gui/src/app/views/utils.js

```diff
@@ -1524,6 +1524,52 @@ YUI.add('juju-view-utils', function(Y) {
   };
 
   /**
+    Check that a value is valid and not null.
+
+    @method generateCloudCredentialName
+    @param {Any} value The value to check
+    @returns {Boolean} Whether the value is not undefined, null or NaN.
+  */
+  utils.isValue = value => {
+    return value !== undefined && value !== null;
+  };
+
+  /**
+    Parse a URL for the querystring and return it as an object.
+
+    @method parseQueryString
+    @param {String} URL The URL to get the query string from.
+    @returns {Object} The parsed query string..
+  */
+  utils.parseQueryString = URL => {
+    const querystring = URL.split('?')[1];
```

Comment:
We should never get double `?` in the url, but we should probably defend against it?
`URL.split('?'); querystring.shift(); querystring.join('?');` Probably unnecessary, but urls are scary things.